### ISSUE TITLE
webasm app run very slowly 

### ISSUE THREAD

**Issue by epowsal** [fyne-io/fyne#5216]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

running slowly

### How to reproduce

edge browser open demo.fyne.io widgets text

### Screenshots

maybe cause by outpu too many:
2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122
wasm_exec.js:22 2024/10/19 15:16:26 Fyne error:  Failed to load image
wasm_exec.js:22 2024/10/19 15:16:26   Cause: open ../../theme/icons/fyne.png: not implemented on js
wasm_exec.js:22 2024/10/19 15:16:26   At: /Users/andy/Code/Fyne/fyne/canvas/image.go:122

### Example code

no

### Fyne version

2.5.1

### Go compiler version

1.23.1

### Operating system and version

windows 11

### Additional Information

no

**Comment by andydotxyz** [comment]
I don't think it relates to the error (though it could!) the WASM -> WebGL upload is very slow and we need to optimise to avoid this.
We may also have accidentally left debug on for the demo site!

**Comment by epowsal** [comment]
Can you made a pure html driver for ui widgets.using gopherjs as event brige. It maybe very fast.

**Comment by andydotxyz** [comment]
We previously did have a gopherJS driver, but the language support is many versions behind so we couldn't keep being held back.

**Comment by adamdecaf** [comment]
I was looking at improving fyne's wasm performance for my application and noticed there are some large in-memory tables storing emojis and UTF-8 tables (from [`yuin/goldmark`](https://github.com/yuin/goldmark)). 

Are these needed by default in fyne wasm apps? I would think their removal could speed up fyne wasm apps until Go can offer better performance.

**Comment by adamdecaf** [comment]
The https://demo.fyne.io/ app seems pretty responsive. Are there any tips that project uses we can port to our own apps? The [fyne app I'm working on](https://github.com/moov-io/watchman/tree/master/internal/ui) is pretty simple.

**Comment by andydotxyz** [comment]
You can disable debug symbols with "fyne serve -release", it can make a big difference. In the next release they will not be included by default.

**Comment by andydotxyz** [comment]
The emoji table is the font loaded to render those characters - you can turn it off what the "no_emoji" tag iirc.
Not sure why goldmark needs a big table but that's for the markdown parsing of TextGrid.
